### ISSUE TITLE
64 bit appimage failure on Fedora 24

### ISSUE THREAD

**Issue by crayxt** [markummitchell/engauge-digitizer#168]
```
[user@linux ~]$ ./digit-exe-universal-linux-64-bit-without-installer-9_1.AppImage 
/tmp/.mount_zQ6ydY
/tmp/.mount_zQ6ydY/usr/bin/engauge.wrapper
/tmp/.mount_zQ6ydY/engauge.desktop
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
process 4388: D-Bus library appears to be incorrectly set up; failed to read machine uuid: Failed to open "/var/lib/dbus/machine-id": No such file or directory
See the manual page for dbus-uuidgen to correct this issue.
  D-Bus not built with -rdynamic so unable to print a backtrace
Aborted (core dumped)
```

Let's cast @probonopd here

**Comment by crayxt** [comment]
Yeah, but Fedora 24 64-bit seems to run just fine without that file.

**Comment by markummitchell** [comment]
I'll try running the appimage in a newly installed Fedora 24 64 bit virtual machine, so I can reproduce the problem and explore fixes.

**Comment by markummitchell** [comment]
I had no problems with Fedora 24 64 bit in VirtualBox. I made a standard desktop install with full upgrade. Then I installed KDE and emacs. Engauge ran fine. Do you have problems with a LiveCD version of Fedora 24?

**Comment by crayxt** [comment]
I have above issue on installed Fedora 24 with GNOME.

**Comment by probonopd** [comment]
Does my workaround from above help in this situation?

**Comment by markummitchell** [comment]
Yes.

There are 3 errors I encountered with Fedora 64:

1) Kernel 4.6.4 (from 'dnf upgrade' just after installation) crashes with kernel panic during operating sytem start up. Using kernel 4.5.5 that came with the Fedora installer worked just fine. This is not an Engauge issue but included in case it helps anyone.

2) Error 'D-Bus library appears to be incorrectly set up; failed to read machine uuid' appears at Engauge startup. As suggested, 'sudo su' followed by 'dbus-uuidgen > /var/lib/dbus/machine-id' fixes this problem. Someone at another site mentioned that dbus-uuidgen utility is part of the 'dbus' package, but that package seems to be loaded by default. Thanks @probonopd!

3) Error 'QCommon::drawComplexControl: Control 1 not handled' appears at Engauge startup. An easy fix is to add '-style=plastique' to the command line. So the command line to run Engauge looks like './digit-exe-universal-linux-64-bit-without-installer-9_1.AppImage -style=plastique'

With these fixes, the Engauge appimage runs fine in Fedora 24.

I'll document this in the release notes, and maybe make code changes if that would help. Suggestions?

**Comment by probonopd** [comment]
At least the -style=plastique could be handled inside the AppImage, e.g., by using an according configuration or startup script.

**Comment by markummitchell** [comment]
Fixed in AppImage 9.1b release.

**Comment by probonopd** [comment]
How?

**Comment by markummitchell** [comment]
Just a workaround, until I learn of a better solution. I first tried to use the Ubuntu solution (#172) which traps the crash at startup (due to the catastrophic gtk library incompatibility) and restarts engauge. However, the gtk library incompatibility with fedora is not catastrophic but only

**Comment by markummitchell** [comment]
The new solution, just checked in, is straightforward. If Fedora is the operating system then (1) default style is changed from gtk which prevents corrupted menus and (2) the /var/lib/dbus/machine-id file is checked and a suggestion is made to the user about how to create that file, which prevents the dbus crash. If there still is a crash, which is probably due to gtk issues, then engauge is restarted with a different default style.

**Comment by crayxt** [comment]
Is this appimage built on Ubuntu? If yes, I'd rather consider migrating to Centos 6 as a base. Ubuntu is known for modifying system libraries for their own needs.

**Comment by markummitchell** [comment]
I've created a set of virtual machines and operating systems for building and testing releases. I've got Centos 6 32 and 64, Ubuntu 32, Kubuntu 64, Windows 7, Windows 10 64, Fedora 64. I've verified the AppImages were created with Centos 32 and 64. This was all triggered by the dropping of support for 32 bit images in (at least) some 64 bit Linux distributions and Windows 10 - it was no longer possible to create 32 bit images for everyone. Now 32 and 64 bit images are released. In the case of Windows, I built 32 bit images in Windows 7 but they did not work in Windows 10. Easiest solution for Windows was just to build for each OS version separately.
